### DISCUSSION TITLE
Platform Specific Package

### DISCUSSION THREAD

**Discussion by rileyhawk1417** [palexdev/MaterialFX#148] [Archived]
Hi love the project and am interested in using it. I just have a question if its possible to platform specific packages.

```xml
        <dependency>
            <groupId>io.github.palexdev</groupId>
            <artifactId>materialfx</artifactId>
            <version>11.13.2</version>
            <classifier>win</classifier>
        </dependency>
```
I noticed that the JavaFX maven package supports "<classifier>" to specify for the platform you are building.

I usually write and build javafx apps on linux then package them using maven. Can MaterialFX carry all the required libraries so it can run on any platfrom right out of the jar file?

**Comment by palexdev** [comment]
MaterialFX has no such thing as platform specific packages.
It already brings all the needed dependencies (such as VirtualizedFX). JavaFX should depend on your project though, as the dependency is transitive on MaterialFX

**Reply by rileyhawk1417** [reply]
Thanks for the answer will give it a try and let you know how it goes 😄

**Reply by rileyhawk1417** [reply]
@palexdev Well tried building on Linux then running it on a Windows Machine. Didnt work out it was looking for d3d and other components. The build system I use is Maven and the commands are:

```bash
mvn clean compile package 
```

Or somtimes I use install 
```bash
mvn clean compile install
```

Any advice on how to get around this problem, or its better to just code and buld on Windows directly?

**Reply by palexdev** [reply]
I don't know about Maven to be honest
But to build MaterialFX Demo for all the major platforms I use jlink
You can check the build.gradle in the demo module to see what's going on

**Reply by rileyhawk1417** [reply]
Ok thanks for the tip will head over and give it a shot. Might add gradle scripts for the build specifically.

**Reply by rileyhawk1417** [reply]
Well after fighting for a while with setting up gradle and the dependancies, got it to work and run outside dev environment. I think gradle would be the best option for most projects, especially when it comes to using the jlink plugin.

**Comment by rileyhawk1417** [comment]
MaterialFX worked better in gradle than maven.
